### PR TITLE
Bulk indexing and usability updates 

### PR DESCRIPTION
**Pull request by schiavo**
Initial load of the i14y drawing is handled by Drupal search setting on cron. 
The admin page for usasearch shows as a tab on the Drupal search configuraton page
Field labels, documentation and other changes to the admin section make the modules easier to use

### CODE REVIEW

File: usasearch_api/usasearch_api.module [hunk 1]
```diff
@@ -201,3 +202,85 @@ function usasearch_api_update_doc_action($node, $context) {
...
+/**
+ * Implements hook_cron().
+ *
+ * @see search_cron()
+ */
+function usasearch_api_cron() {
+  // We register a shutdown function to ensure that search_total is always up
+  // to date.
+  drupal_register_shutdown_function('search_update_totals');
+dpm('got cron?');
```
Comment (grasmash) [inline usasearch_api/usasearch_api.module:238]: Please remove dpm() calls.

File: usasearch_api/usasearch_api.module [hunk 2]
```diff
@@ -201,3 +202,85 @@ function usasearch_api_update_doc_action($node, $context) {
...
+  foreach (variable_get('search_active_modules', array('node', 'user')) as $module) {
+    // Update word index
+    $module == 'node' ? module_invoke('usasearch_api', 'update_index') : module_invoke($module, 'update_index');
+  }
+}
+
+// function usasearch_api_search_info() {
```
Comment (grasmash) [inline usasearch_api/usasearch_api.module:245]: Please remove unused code.

File: usasearch.module [hunk 3]
```diff
@@ -10,12 +10,27 @@
 function usasearch_menu() {
   $items = array();
 
+  // definition for additional tab.
```
Comment (grasmash) [inline usasearch.module:13]: Please run this PR through code sniffer.

File: usasearch_api/usasearch_api.module [hunk 4]
```diff
@@ -124,6 +124,7 @@ function usasearch_api_convert_node_to_document($node) {
 /**
  * Implements hook_form_FORM_ID_alter().
  */
+// TODO decide whether to move this to main module
```
Comment (grasmash) [inline usasearch_api/usasearch_api.module:127]: Good idea.

File: usasearch_api/usasearch_api.module [hunk 5]
```diff
@@ -201,3 +202,85 @@ function usasearch_api_update_doc_action($node, $context) {
...
+    unset($implementations['search']);
+  }
+}
+
+/**
+ * implimentation of hook_form_alter()
+ */
+function usasearch_api_form_search_admin_settings_alter(&$form, &$form_state, $form_id){
+  $items = drupal_map_assoc(array(10, 20, 50, 100, 200, 500,1000,10000));
+  $form['indexing_throttle']['search_cron_limit'] = array(
```
Comment (grasmash) [inline usasearch_api/usasearch_api.module:220]: Please namespace variable name.

File: usasearch_api/usasearch_api.module [hunk 6]
```diff
@@ -201,3 +202,85 @@ function usasearch_api_update_doc_action($node, $context) {
...
+    '#type' => 'select',
+    '#title' => t('Number of items to index per cron run'),
+    '#default_value' => variable_get('search_cron_limit', 100),
+    '#options' => $items,
+    '#description' => t('The maximum number of items indexed in each pass of a <a href="@cron">cron maintenance task</a>. If necessary, reduce the number of items to prevent timeouts and memory errors while indexing.', array('@cron' => url('admin/reports/status')))
```
Comment (grasmash) [inline usasearch_api/usasearch_api.module:225]: `<a href="@cron">cron maintenance task</a>` should be replaced with @cron_link, which should be generated by the l() function.

File: usasearch_api/usasearch_api.module [hunk 7]
```diff
@@ -201,3 +202,85 @@ function usasearch_api_update_doc_action($node, $context) {
...
+
+/**
+ * Implements hook_update_index().
+ *
+ * @see node_update_index()
+ */
+function usasearch_api_update_index() {
+
+  $limit = (int)variable_get('search_cron_limit', 100);
+  $result = db_query_range("SELECT n.nid FROM {node} n LEFT JOIN {search_dataset} d ON d.type = 'node' AND d.sid = n.nid WHERE d.sid IS NULL OR d.reindex <> 0 ORDER BY d.reindex ASC, n.nid ASC", 0, $limit);
```
Comment (grasmash) [inline usasearch_api/usasearch_api.module:261]: Why is Drupal's search index being used at all?

Comment (schiavo) [inline usasearch_api/usasearch_api.module:261]: It's used to track what's been indexed. Perhaps there's a way to do this without the overhead

**Comment by grasmash** [comment]
This was merged.

**Pull request closed without merging**